data/src/main/java/org/apache/iceberg/data/BaseDeleteLoader.java

```diff
@@ -146,6 +151,36 @@ private <T> Iterable<T> materialize(CloseableIterable<T> iterable) {
   @Override
   public PositionDeleteIndex loadPositionDeletes(
       Iterable<DeleteFile> deleteFiles, CharSequence filePath) {
+    if (containsDVs(deleteFiles)) {
+      DeleteFile deleteFile = Iterables.getOnlyElement(deleteFiles);
```

Comment:
minor: should we add some validation to make sure `deleteFiles` indeed only has a single entry?